### PR TITLE
feat: allow creating tables from a source query

### LINKED ISSUE #1156: CREATE TABLE ... AS ... support
## Summary

Create table from query

Since we have native tables now, might as well add this in too.

## Rationale

Reading parquet files or json files from data sets

### PR DESCRIPTION
**Pull request by adhish20**
Fixes #1156

### CODE REVIEW

File: testdata/sqllogictests/table.slt [hunk 1]
```diff
@@ -19,3 +19,6 @@ create table table_with_list (a text[])
 
 statement ok
 insert into table_with_list values (['hello', 'world'])
+
+statement ok
+create table t1 as select * from generate_series(1, 5, 2);
```
Comment (vrongmeal) [inline testdata/sqllogictests/table.slt:24]:
```suggestion

statement ok
create table t1 as select * from generate_series(1, 5, 2);

query I
select * from t1;
----
1
3
5
```

File: crates/sqlexec/src/context.rs [hunk 2]
```diff
@@ -260,6 +263,34 @@ impl SessionContext {
...
+                .insert_into(self.get_df_state(), physical)
+                .await?;
+
+            let context = self.task_context();
+            let mut stream = execute_stream(physical, context)?;
+
+            // Drain the stream to actually "write" successfully.
+            while let Some(res) = stream.next().await {
+                let _ = res?;
+            }
```
Comment (vrongmeal) [inline crates/sqlexec/src/context.rs:291]: Can we add a helper function and call it both for `insert` and here?

**Review by vrongmeal — APPROVED** [review]
Looks good to me — just a couple of minor comments.

File: crates/sqlexec/src/context.rs (hunk 2 above)
Comment (adhish20) [reply crates/sqlexec/src/context.rs:291]: sure. I will update that

**Review by vrongmeal — APPROVED** [review]
🚀

**Pull request merged**